src/ResourceManager/Sql/Commands.Sql/VirtualNetworkRule/Cmdlet/RemoveAzureSqlServerVirtualNetworkRule.cs

```diff
@@ -0,0 +1,89 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.VirtualNetworkRule.Cmdlet
+{
+    /// <summary>
+    /// Defines the Remove-AzureRmSqlServerVirtualNetworkRule cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmSqlServerVirtualNetworkRule", SupportsShouldProcess = true, ConfirmImpact = ConfirmImpact.Medium)]
+    public class RemoveAzureSqlServerVirtualNetworkRule : AzureSqlServerVirtualNetworkRuleCmdletBase
+    {
+        /// <summary>
+        /// Gets or sets the name of the VirtualNetwork rule to remove
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Azure Sql Server Virtual Network Rule name")]
+        [ValidateNotNullOrEmpty]
+        public string VirtualNetworkRuleName { get; set; }
+
+        /// <summary>
+        /// Defines whether it is ok to skip the requesting of rule removal confirmation
+        /// </summary>
+        [Parameter(HelpMessage = "Skip confirmation message for performing the action")]
+        public SwitchParameter Force { get; set; }
+
+        /// <summary>
+        /// Gets the entity to delete
+        /// </summary>
+        /// <returns>The entity going to be deleted</returns>
+        protected override IEnumerable<Model.AzureSqlServerVirtualNetworkRuleModel> GetEntity()
+        {
+            return new List<Model.AzureSqlServerVirtualNetworkRuleModel>() {
+                ModelAdapter.GetVirtualNetworkRule(this.ResourceGroupName, this.ServerName, this.VirtualNetworkRuleName)
+            };
+        }
+
+        /// <summary>
+        /// Apply user input.  Here nothing to apply
+        /// </summary>
+        /// <param name="model">The result of GetEntity</param>
+        /// <returns>The input model</returns>
+        protected override IEnumerable<Model.AzureSqlServerVirtualNetworkRuleModel> ApplyUserInputToModel(IEnumerable<Model.AzureSqlServerVirtualNetworkRuleModel> model)
+        {
+            return model;
+        }
+
+        /// <summary>
+        /// Deletes the server.
+        /// </summary>
+        /// <param name="entity">The server being deleted</param>
+        /// <returns>The server that was deleted</returns>
+        protected override IEnumerable<Model.AzureSqlServerVirtualNetworkRuleModel> PersistChanges(IEnumerable<Model.AzureSqlServerVirtualNetworkRuleModel> entity)
+        {
+            ModelAdapter.RemoveVirtualNetworkRule(this.ResourceGroupName, this.ServerName, this.VirtualNetworkRuleName);
+            return entity;
+        }
+
+        /// <summary>
+        /// Entry point for the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            if (!Force.IsPresent && !ShouldProcess(
```

Comment:
This is redundant and not needed. See: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Sql/Commands.Sql/Common/AzureSqlCmdletBase.cs#L124

Reply:
I was asked by the Powershell team to add this when they reviewed my cmdlets.